Assets/_BForBoss/_Entities/Scripts/LifeCycle.cs

```diff
@@ -46,6 +46,11 @@ public void HealBy(float amount)
             OnHeal?.Invoke();
             _currentHealth = Mathf.Min(_maxHealth, _currentHealth + amount);
         }
+        
+        public void NotifyOnDeath(Action onDeathCallback)
```

Comment:
NiT - can the default value of the argument be nil?

Reply:
Hmm good call, cant use the `?` operator before Invoke, so will just wrap the parameter in a null check

Reply:
Can it be set like this? 
`Action onDeathCallback = null`